smqtk_classifier/impls/classify_descriptor_supervised/sklearn_svm.py

```diff
@@ -0,0 +1,449 @@
+
+    :param svm_model_uri: Path to the libSVM model file.
+    :param svm_label_map_uri: Path to the pickle file containing this
+        model's output labels.
+    :param train_params: SVM parameters used for training. See libSVM
+        documentation for parameter flags and values. A parameter with no value
+        on the SVM CLI should be given an empty string value,
+        E.g. `{"-q": ""}`.
+    :param normalize: Normalize input vectors to training and
+        classification methods using ``numpy.linalg.norm``. This may either
+        be  ``None``, disabling normalization, or any valid value that
+        could be passed to the ``ord`` parameter in ``numpy.linalg.norm``
+        for 1D arrays. This is ``None`` by default (no normalization).
+    :param n_jobs:
+        Number of processes to use to parallelize prediction. If None or a
+        negative value, all cores are used.
+    """
+
+    # noinspection PyDefaultArgument
+    def __init__(
+        self,
+        svm_model_uri: Optional[str] = None,
+        svm_label_map_uri: Optional[str] = None,
+        train_params: SVM_PARAM_MAPPING_T = {
+            '-s': 0,  # C-SVC, assumed default if not provided
+            '-t': 0,  # linear kernel
+            '-b': 1,  # enable probability estimates
+            '-c': 2,  # SVM parameter C
+        },
+        normalize: Optional[Union[int, float, str]] = None,
+        n_jobs: Optional[int] = 4,
+    ):
+        super(SkLearnSvmClassifier, self).__init__()
+
+        self.svm_model_uri = svm_model_uri
+        self.svm_label_map_uri = svm_label_map_uri
+
+        # Elements will be None if input URI is None
+        #: :type: None | smqtk.representation.DataElement
+        self.svm_model_elem = \
+            svm_model_uri and from_uri(svm_model_uri)
+        #: :type: None | smqtk.representation.DataElement
+        self.svm_label_map_elem = \
+            svm_label_map_uri and from_uri(svm_label_map_uri)
+
+        # Shallow copy to shield from modifying input.
+        self.train_params = dict(train_params)
+        self.normalize = normalize
+        self.n_jobs = n_jobs
+        # Validate normalization parameter by trying it on a random vector
+        if normalize is not None:
+            self._norm_vector(numpy.random.rand(8))
+
+        # generated parameters
+        self.svm_model: Optional[svm.svm_model] = None
+        # dictionary mapping SVM integer labels to semantic labels
+        self.svm_label_map: Dict[int, Hashable] = {}
+
+        self._reload_model()
+
+    @classmethod
+    def is_usable(cls) -> bool:
+        return None not in {scipy, svm}
+
+    def get_config(self) -> Dict[str, Any]:
+        return {
+            "svm_model_uri": self.svm_model_uri,
+            "svm_label_map_uri": self.svm_label_map_uri,
+            "train_params": self.train_params,
+            "normalize": self.normalize,
+            "n_jobs": self.n_jobs,
+        }
+
+    def __getstate__(self) -> Any:
```

Comment:
The use of this an `__setstate__` are likely not needed in this case due not wrapping a C-library's handle pointer any more. I assume the wrapped scikit-learn model instance knows how to un/pickle itself.